### PR TITLE
POC: Use pnpm or yarn pnpm

### PR DESCRIPTION
**Pull request by joshhunt**
See https://github.com/grafana/grafana/pull/76643 for full explaination.

Based off that branch, but adds support for pnpm as well. Both `yarn install` and `pnpm install` should work in this branch.

Removed the `@storybook/blocks` patch because it was causing an exception why trying to pnpm install. Will have to figure out solution to that (I vote to just stop patching and deal with it)

~~`pnpm jest public/app/plugins/panel/geomap/utils/measure.test.ts` passes, but `pnpm jest browse-dashboards` does not - something about "Cannot find module 'slate-dev-environment'", which seems relevent to the patches we apply.~~

### CODE REVIEW

**Comment by joshhunt** [comment]
pnpm start doesn't currently build:

```
> yarn start:noLint
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module '/home/joshhunt/dev/grafana/grafana/node_modules/.store/esbuild-npm-0.18.12-cca10dbb26/node_modules/esbuild/bin/esbuild'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.17.1
no i18n compile yet, all good
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module '/home/joshhunt/dev/grafana/grafana/node_modules/.store/webpack-virtual-d7a2d4d897/node_modules/webpack/bin/webpack.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.17.1
```

**Comment by joshhunt** [comment]
Fixed tests failing and `yarn start` not building in https://github.com/grafana/grafana/pull/76650/commits/295a687186be0d605b77d19b1dec22d8d105d6af. Basically just needed to tell pnpm about the patches.

File: package.json [hunk 1]
```diff
@@ -423,14 +422,26 @@
     "@types/slate": "0.47.11",
     "@rushstack/rig-package": "0.5.1",
     "@rushstack/ts-command-line": "4.16.1",
-    "ngtemplate-loader/loader-utils": "^2.0.0",
```
Comment (joshhunt) [inline package.json:426]: While pnpm does support both the resolutions and pnpm.overrides fields, pnpm chokes on this syntax for overriding transitive dependencies, so I've removed this for now.

I don't think we actually build angular templates any more, so it probably doesn't hurt to just remove this.

File: package.json [hunk 2]
```diff
@@ -423,14 +422,26 @@
...
     "slate-dev-environment@^0.2.2": "patch:slate-dev-environment@npm:0.2.5#.yarn/patches/slate-dev-environment-npm-0.2.5-9aeb7da7b5.patch",
     "react-split-pane@0.1.92": "patch:react-split-pane@npm:0.1.92#.yarn/patches/react-split-pane-npm-0.1.92-93dbf51dff.patch",
     "@storybook/blocks@7.4.5": "patch:@storybook/blocks@npm%3A7.4.5#./.yarn/patches/@storybook-blocks-npm-7.4.5-5a2374564a.patch"
   },
+  "pnpm": {
+    "overrides": {
+      "ngtemplate-loader>loader-utils": "^2.0.0",
+      "slate-dev-environment@^0.2.2": "^0.2.2",
+      "react-split-pane@0.1.92": "0.1.92",
+      "@storybook/blocks@7.4.5": "7.4.5"
```
Comment (joshhunt) [inline package.json:437]: These overrides are required to "undo" the yarn resolutions. The patches are applied by different configuration.

If we switch fully to pnpm, we could just remove these from both the resolutions and pnpm.overrides objects.

**Comment by joshhunt** [comment]
Interesting note from CI: _current_ Drone/cache configuration is not supported with at least yarn pnpm. https://drone.grafana.net/grafana/grafana/145401/1/3

> `[Error: EXDEV: cross-device link not permitted, link '/root/.yarn/berry/index/73/737372d0ce81cdc670ebd131771ff8089e86dd99.dat' -> '/drone/src/node_modules/.store/@jridgewell-resolve-uri-npm-3.1.0-6ff2351e61/package/LICENSE'] {`

Have not ran pnpm in CI yet, so this may also be problematic for pnpm.

**Comment by leeoniya** [comment]
there is also https://bun.sh/docs/cli/install

**Comment by joshhunt** [comment]
Indeed @leeoniya! I think it's unlikely that we will choose bun's package manager as the supported package manager for grafana, but if we do the work to restore a node_modules folder (ie switch to pnpm) I would be keen to see what is required to also make bun work. Bun could be a "secret unsupported option" for developers to use locally.

**Pull request closed without merging**